rust/src/csv/reader.rs

```diff
@@ -78,16 +197,41 @@ impl Reader {
         let csv_reader = csv::ReaderBuilder::new()
             .has_headers(has_headers)
             .from_reader(BufReader::new(file));
-
         let record_iter = csv_reader.into_records();
         Reader {
-            schema: schema.clone(),
+            schema: schema,
             projection,
             record_iter,
             batch_size,
         }
     }
 
+    pub fn from_builder(file: File, builder: ReaderBuilder) -> Self {
+        // when no schema is provided, the CSV should have headers
```

Comment:
How do we know that CSV should have headers when no schema is provided? should this be a pre-condition check?

Reply:
I've removed this logic, and now instead generate default column names (`column_1`, `column_2`, ...) when headers aren't provided. So we no longer need a pre-condition check.